bridgekeeper/rules.py

```diff
@@ -122,7 +122,20 @@ def check(self, user, instance=None):
         this rule for a given instance, or if none is provided,
         every instance.
         """
-        raise NotImplementedError()
+        # If an instance context was not given, we could only possibly return
+        # True if the rule universally passes for the user.
+        if self.query(user) is UNIVERSAL:
+            return True
+        elif instance is None:
+            # An instance context was not given, and the rule does not
+            # universally pass. We must return false.
+            return False
+        # Get a queryset with only this instance in it
+        queryset = instance._meta.model.objects.filter(pk=instance.pk)
+        # Filter the queryset by the Rule
+        queryset = self.filter(user, queryset)
+        # Check if it exists
+        return queryset.exists()
```

Comment:
I'm not sure if it's worth documenting here but we could have a `elif self.query(user) is EMPTY` check here as well, but that's actually unnecessary because if `instance` is `None` then we return `False` anyway, and if `instance` is not `None`, the call to `filter()` adds `none()` to the queryset which makes the later `exists()` call return `False` with no database queries anyway. So adding it would not optimise anything.

Another option could be to do:
```
        if instance is None:
            # An instance context was not given, and the rule does not
            # universally pass. We must return false.
            return self.query(user) is UNIVERSAL
        # Get a queryset with only this instance in it
        queryset = instance._meta.model.objects.filter(pk=instance.pk)
        # Filter the queryset by the Rule
        filtered_queryset = self.filter(user, queryset)
        # Check if it exists
        return queryset == filtered_queryset or filtered_queryset.exists()
```

I don't know if that reads cleaner or not.

Reply:
I had a similar thought and determined that it wouldn't make a difference, but I think you're right in saying that it could be made more explicit. 

I've seen a couple of mailing list threads float by that made me nervous about equating QuerySets so I might opt to just check for `EMPTY`.